### PR TITLE
[DEV-6238] Advanced Search Hash Url (a) Loads search and (b) doesn't generate duplicate hash

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
- On Advanced Search in QAT, the hash url page (`/search/234qkd234`) was not loading the search
- On Advanced search in PROD, the hash url page was generating a duplicate hash for the same search

**Technical details:**
changing useEffect conditional logic:

- caller of generateHash updated to look at appliedFilters property in redux for source of truth
- caller of history.replace() updated to look at new previousProps ref & areAppliedFiltersEmpty prop in redux for source of truth

**JIRA Ticket:**
[DEV-6238](https://federal-spending-transparency.atlassian.net/browse/DEV-6238)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
`N/A`Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
`N/A`Verified mobile/tablet/desktop/monitor responsiveness
`N/A`Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A`[API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/js/containers/search/SearchContainer.jsx [hunk 1]
```diff
@@ -86,6 +86,13 @@ const SearchContainer = ({ history }) => {
     const [downloadInFlight, setDownloadInFlight] = useState(false);
     const [generateHashInFlight, setGenerateHashInFlight] = useState(false);
     const request = useRef(null);
+    const areAppliedFiltersEmptyRef = useRef();
+
+    useEffect(() => {
+        areAppliedFiltersEmptyRef.current = areAppliedFiltersEmpty;
+    });
```
Comment (jonhill13) [inline src/js/containers/search/SearchContainer.jsx:93]: We should only update this if areAppliedFiltersEmpty changes right?

File: tests/containers/search/SearchContainer-test.jsx [hunk 2]
```diff
@@ -2,46 +2,51 @@
...
 // mock the child component by replacing it with a function that returns a null element
-jest.mock('components/search/SearchPage', () =>
-    jest.fn(() => null));
+jest.mock('components/search/SearchPage', () => (
+    jest.fn(() => null)
+));
 
 jest.mock('helpers/searchHelper', () => require('./filters/searchHelper'));
-jest.mock('helpers/fiscalYearHelper', () => require('./filters/fiscalYearHelper'));
+// jest.mock('helpers/fiscalYearHelper', () => require('./filters/fiscalYearHelper'));
```
Comment (jonhill13) [inline tests/containers/search/SearchContainer-test.jsx:25]: Do we still want this?

**Review by jonhill13 — CHANGES_REQUESTED** [review]

File: src/js/containers/search/SearchContainer.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/containers/search/SearchContainer.jsx:93]: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state

Comment (jonhill13) [reply src/js/containers/search/SearchContainer.jsx:93]: Yes, Thank you.  Question still stands. This `useEffect` runs anytime the component updates, does the `ref` current need to be updated from `false` to `false` or `true` to `true`?

**Review by jonhill13 — APPROVED** [review]

**Pull request merged**